### ISSUE TITLE
Facebook login

### ISSUE THREAD

**Issue by crysfel** [adamjmcgrath/react-native-simple-auth#5]
Hello!

I have followed the steps on the Read.me, the project builds and runs correctly. However when doing the login with Facebook I receive an error on the simulator.

```
2015-07-22 22:49:23.014 [info][tid:com.facebook.React.JavaScript] 'Running application "CreedProfetas" with appParams: {"rootTag":1,"initialProps":{}}. __DEV__ === false, development-level warning are OFF, performance optimizations are ON'
2015-07-22 22:49:23.019 CreedProfetas[11616:1217639] Configuring facebook with {
    "app_id" = ************328;
}
2015-07-22 22:49:25.691 CreedProfetas[11616:1217580] 
Response: (null)
Error:Error Domain=SimpleAuthErrorDomain Code=102 "The operation couldn’t be completed. (SimpleAuthErrorDomain error 102.)" UserInfo=0x7fad825252f0 {NSUnderlyingError=0x7fad84bc1330 "The operation couldn’t be completed. (com.apple.accounts error 6.)"}
2015-07-22 22:49:25.697 [info][tid:com.facebook.React.JavaScript] { description: 'The operation couldn’t be completed. (com.apple.accounts error 6.)',
  code: 102 }
```

I'm not sure what the 102 error is. Anyway, when I run the app in a real device I receive a different error.

```
2015-07-22 22:39:30.887 [info][tid:com.facebook.React.JavaScript] 'Running application "CreedProfetas" with appParams: {"rootTag":1,"initialProps":{}}. __DEV__ === false, development-level warning are OFF, performance optimizations are ON'
2015-07-22 22:39:30.908 CreedProfetas[624:270277] Configuring facebook with {
    "app_id" = ************328;
}
2015-07-22 22:39:51.180 CreedProfetas[624:270346] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** setObjectForKey: object cannot be nil (key: first_name)'
*** First throw call stack:
(0x1831842d8 0x1949500e4 0x18306d428 0x100278808 0x100278310 0x18318a100 0x1830862fc 0x18308ae30 0x10033c658 0x10038d43c 0x10038ccf4 0x10038612c 0x100394968 0x100359784 0x100372fd8 0x100394968 0x100359784 0x100385c34 0x100394968 0x10035c9f8 0x100395868 0x10035c934 0x1003896ec 0x100385ae4 0x100386164 0x100394968 0x100359784 0x10036c904 0x10036cbf4 0x100394968 0x100359784 0x100385c34 0x100394968 0x10035c9f8 0x10035e96c 0x10035a63c 0x1002bcfd4 0x1002bcf94 0x1002c7db8 0x1002c02c4 0x1002ca5d4 0x1002cc248 0x19518122c 0x195180ef0)
libc++abi.dylib: terminating with uncaught exception of type NSException
Printing description of [0]:
(struct __lldb_autogen_nspair) [0] = {
  key = 0x0000000174225720
  value = 0x000000017442f0c0
}
Printing description of [1]:
(struct __lldb_autogen_nspair) [1] = {
  key = 0x0000000174224020
  value = 0x0000000174059f50
}
```

It's failing when is trying to read the last name on the remoteAccount object, the last name doesn't exist, basically all this object has is the name key, everything else is missing.

SimpleAuthFacebookProvider.m

```
- (NSDictionary *)infoDictionaryWithRemoteAccount:(NSDictionary *)remoteAccount systemAccount:(ACAccount *)systemAccount {
    NSMutableDictionary *dictionary = [[NSMutableDictionary alloc] init];

    dictionary[@"name"] = remoteAccount[@"name"];
    dictionary[@"first_name"] = remoteAccount[@"first_name"];
    dictionary[@"last_name"] = remoteAccount[@"last_name"];
    dictionary[@"verified"] = remoteAccount[@"verified"] ?: @NO;

    id email = remoteAccount[@"email"];
    ...
```

Any help would be appreciate.

Regards

**Comment by crysfel** [comment]
The error on the simulator is happening because I'm not signed into facebook, however once I signed in I'm getting the same error as in the real device. Apparently the remoteAccount object only contains the name key, all other keys are missing and therefore I'm getting this error.

Any thoughts on this? Thanks

**Comment by adamjmcgrath** [comment]
Hi, I suspect this might be because of the v2.4 Graph API changes:

> Fewer default fields for faster performance: To help improve performance on mobile network connections, we've reduced the number of fields that the API returns by default. You should now use the ?fields=field1,field2 syntax to declare all the fields you want the API to return.

`first_name` may no longer be a default field.

If that's the case this would be an issue with [SimpleAuth](/calebd/SimpleAuth), we may need to explicitly add the fields we need to the request in https://github.com/calebd/SimpleAuth/blob/master/Pod/Providers/Facebook/SimpleAuthFacebookProvider.m#L67

Something like:

```
NSURL *URL = [NSURL URLWithString:@"https://graph.facebook.com/me?fields=first_name"];
```

I'll take a look this weekend.

**Comment by adamjmcgrath** [comment]
Created https://github.com/calebd/SimpleAuth/pull/109

**Comment by beinginorbit** [comment]
@adamjmcgrath What's the status of this issue? Auth with react-native is a roller coaster ride, like a bad wooden one.

**Comment by adamjmcgrath** [comment]
Still blocked by calebd/SimpleAuth#109 I'm afraid, you can use my fork directly as a workaround:

Syntax in your Podfile goes something like this:

`pod 'SimpleAuth/Facebook', :git => 'https://github.com/adamjmcgrath/SimpleAuth-1.git', :branch => 'facebook-2.4-breaking-changes'`

**Comment by adamjmcgrath** [comment]
This is resolved in v2.0.0 as I no longer depend on SimpleAuth
